pkg/apis/core/validation/validation.go

```diff
@@ -2589,9 +2589,25 @@ func validateLifecycle(lifecycle *core.Lifecycle, fldPath *field.Path) field.Err
 	if lifecycle.PreStop != nil {
 		allErrs = append(allErrs, validateHandler(lifecycle.PreStop, fldPath.Child("preStop"))...)
 	}
+	if lifecycle.Type != nil {
+		allErrs = append(allErrs, validateLifecycleType(lifecycle.Type, fldPath.Child("type"))...)
+	}
 	return allErrs
 }
 
+var supportedLifecycleTypes = sets.NewString(string(core.LifecycleTypeSidecar))
+
+func validateLifecycleType(lifecycleType *core.LifecycleType, fldPath *field.Path) field.ErrorList {
```

Comment:
Do we need to error is isInitContainer && has sidecar?  The API comments say that is an error

Reply:
having the `Lifecycle` field is already a validation error for init containers
```go
	if ctr.Lifecycle != nil {
			allErrs = append(allErrs, field.Invalid(idxPath.Child("lifecycle"), ctr.Lifecycle, "must not be set for init containers"))
		}
```
so I wasn't sure if we needed the extra validation step for `Lifecycle.Type`

Reply:
ACK